File: 0989-add-to-array-form-of-integer/0989-add-to-array-form-of-integer.java
class Solution {
    public List<Integer> addToArrayForm(int[] num, int k) {
        List<Integer> ans = new ArrayList<>();

        int pointer = num.length - 1;

        int carry = 0;

        while (pointer >= 0 || k > 0) {
            int lastArrVal = 0;

            if (pointer >= 0) {
                lastArrVal = num[pointer];
            }

            int lastKDigit = k % 10; // It is last digit from k

            int total = lastArrVal + lastKDigit + carry;

            int sum = total % 10;

            ans.add(sum);

            carry = total / 10;

            pointer--; // Move the pointer to left

            k /= 10; // Remove last digit from the k
        }

        if (carry > 0) {
            ans.add(carry); // Add extra carry to the array list
        }

        Collections.reverse(ans); // Reverse the answer array list

        return ans;
    }
}

File: 0989-add-to-array-form-of-integer/0989-add-to-array-form-of-integer.js
/**
 * @param {number[]} num
 * @param {number} k
 * @return {number[]}
 */
var addToArrayForm = function (num, k) {
    let ans = [];

    let pointer = num.length - 1;

    let carry = 0;

    while (pointer >= 0 || k > 0) {
        let arrLastVal = 0;

        if (pointer >= 0) {
            arrLastVal = num[pointer];
        }

        let lastKDigit = k % 10;

        let total = arrLastVal + lastKDigit + carry;

        let sum = total % 10;

        ans.push(sum);

        carry = Math.trunc(total / 10);

        pointer--;

        k = Math.floor(k / 10);
    }

    if (carry > 0) {
        ans.push(carry);
    }

    ans = ans.reverse();

    return ans;
};